File: mac/Sources/DMShot/UpdateHint.swift
import Foundation

/// Maps the updater state to the version shown by the ACTIVE update hint
/// (status-icon badge + first menu item, spec 2026-07-05). Only states where
/// the user can actually act (update available / restart pending) hint;
/// transient download/extract states keep the hint from their preceding
/// `available` via `readyToInstall`'s version.
enum UpdateHint {
    static func version(for state: UpdateState) -> String? {
        switch state {
        case .available(let version, _): return version
        case .readyToInstall(let version): return version
        case .disabled, .idle, .checking, .upToDate, .downloading, .extracting, .error:
            return nil
        }
    }
}

File: windows/DMShot/Update/UpdateHint.cs
namespace DMShot.Update;

/// <summary>
/// Maps the updater state to the version shown by the ACTIVE update hint
/// (tray-icon badge + first menu item, spec 2026-07-05; mac parity:
/// UpdateHint.swift). Only states where the user can act hint.
/// </summary>
public static class UpdateHint
{
    public static string? VersionFor(UpdateState state)
        => state.Status is UpdateStatus.Available or UpdateStatus.ReadyToInstall
            ? state.Version
            : null;
}
